otelconf/internal/provider/envprovider.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package provider // import "go.opentelemetry.io/contrib/otelconf/internal/provider"
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+const ValidationPattern = `^[a-zA-Z_][a-zA-Z0-9_]*$`
+
+var validationRegexp = regexp.MustCompile(ValidationPattern)
+
+func ReplaceEnvVars(input []byte) ([]byte, error) {
+	re := regexp.MustCompile(`\$\{([a-zA-Z_][a-zA-Z0-9_]*[-]?.*)\}`)
+
+	replaceEnvVars := func(input []byte) ([]byte, error) {
+		var err error
+		out := re.ReplaceAllFunc(input, func(s []byte) []byte {
+			match := re.FindSubmatch(s)
+			if len(match) < 2 {
+				return s
+			}
+			var data []byte
+			data, err = replaceEnvVar(string(match[1]))
+			return data
+		})
+		return out, err
+	}
+	return replaceEnvVars(input)
+}
+
+func replaceEnvVar(uri string) ([]byte, error) {
+	envVarName, defaultValuePtr := parseEnvVarURI(uri)
+	if !validationRegexp.MatchString(envVarName) {
+		return nil, fmt.Errorf("invalid environment variable name: %s", envVarName)
+	}
+
+	val, exists := os.LookupEnv(envVarName)
+	if !exists {
+		if defaultValuePtr != nil {
+			val = *defaultValuePtr
+		}
+	}
```

Comment:
```suggestion
	val := os.Getenv(envVarName)
	if val == "" && defaultValuePtr != nil {
		val = *defaultValuePtr
	}
```

From https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#parsing-empty-value

> The SDK MUST interpret an empty value of an environment variable the same way as when the variable is unset.

Reasons are here: https://github.com/open-telemetry/opentelemetry-specification/pull/2045.